addons/mail/static/src/components/activity_mark_done_popover/activity_mark_done_popover_tests.js

```diff
@@ -134,6 +134,33 @@ QUnit.test('activity with force next mark done popover simplest layout', async f
     );
 });
 
+QUnit.test('activity mark done popover close on ESCAPE', async function (assert) {
+    assert.expect(3);
+
+    await this.start();
+    const activity = this.env.models['mail.activity'].create({
+        canWrite: true,
+        category: 'not_upload_file',
+        id: 12,
+    });
+    await this.createActivityMarkDonePopoverComponent(activity);
+    function onPopoverClose(ev) {
+        assert.step('event_triggered');
+    }
+    document.addEventListener('o-popover-close', onPopoverClose);
+
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover',
+        "Popover component should be present"
+    );
+    const ev = new window.KeyboardEvent('keydown', { bubbles: true, key: "Escape" });
+    document.querySelector(`.o_ActivityMarkDonePopover`).dispatchEvent(ev);
+    assert.verifySteps(['event_triggered'], 'ESCAPE pressed should trigger the right event');
+
+    document.removeEventListener('o-popover-close', onPopoverClose);
```

Comment:
I see another test also using `o-popover-close` on `document` to make assertion on popover.
Cannot use something better, like see in DOM if the popover is (not) present?

Reply:
I think it's made like this to avoid leak how popovers are done. For example if popover preload the content and just hides it on close (or something more tricky like hide data and remove from DOM like 5 seconds later), then the test would fail even if the GUI is ok as the idea is that popover is closed.

Reply:
I think this made sense with jquery popovers, but now with OWL popovers, I think we could trust the DOM. Right?